src/control/go.mod

```diff
@@ -1,9 +1,11 @@
 module github.com/daos-stack/daos/src/control
 
+// NB: When updating minimum Go build version, don't forget to update:
+// - rpm packaging version checks: utils/rpms/daos.spec
+// - debian packaging version checks: debian/control
+// Scons uses this file to extract the minimum version.
 go 1.21
 
-toolchain go1.22.3
-
```

Comment:
I'm not familiar with go, what does this part of the change do?

Reply:
Yeah, the go mod tools added that line automatically, but I don't think we need it. It seems that it's only relevant if we want to use a different toolchain version than the minimum go version.